tests/rules/no-effect-decorator-and-creator.test.ts

```diff
@@ -1,40 +1,77 @@
-import { stripIndent } from 'common-tags'
+import { stripIndents } from 'common-tags'
 import { fromFixture } from 'eslint-etc'
 import path from 'path'
+import type { MessageIds } from '../../src/rules/effects/no-effect-decorator-and-creator'
 import rule, {
-  messageId,
+  noEffectDecoratorAndCreator,
+  noEffectDecoratorAndCreatorSuggest,
 } from '../../src/rules/effects/no-effect-decorator-and-creator'
 import { ruleTester } from '../utils'
 
 ruleTester().run(path.parse(__filename).name, rule, {
   valid: [
-    stripIndent`
-      @Injectable()
-      export class FixtureEffects {
-        creator = createEffect(() => this.actions)
-
-        constructor(private actions: Actions){}
-      }`,
-    stripIndent`
-      @Injectable()
-      export class FixtureEffects {
-        @Effect({ dispatch: false })
-        decorator = this.actions
-
-        constructor(private actions: Actions){}
-      }`,
+    `
+    @Injectable()
+    export class FixtureEffects {
+      creator = createEffect(() => this.actions)
+      constructor(private actions: Actions) {}
+    }`,
+    `
+    @Injectable()
+    export class FixtureEffects {
+      @Effect({ dispatch: false })
+      decorator = this.actions
+      constructor(private actions: Actions) {}
+    }`,
   ],
   invalid: [
     fromFixture(
-      stripIndent`
+      stripIndents`
         @Injectable()
         export class FixtureEffects {
-          @Effect({ dispatch: false })
+          @Effect()
           both = createEffect(() => this.actions)
-          ~~~~ [${messageId}]
-
-          constructor(private actions: Actions){}
+          ~~~~ [${noEffectDecoratorAndCreator}]
+          constructor(private actions: Actions) {}
         }`,
+      {
+        output: stripIndents`
+          @Injectable()
+          export class FixtureEffects {
+
+            both = createEffect(() => this.actions)
+            constructor(private actions: Actions) {}
+          }`,
+      },
     ),
+    {
+      code: stripIndents`
+        @Injectable()
+        export class FixtureEffects {
+          @Effect({ dispatch: false })
+          both = createEffect(() => this.actions)
+          constructor(private actions: Actions) {}
+        }`,
+      errors: [
+        {
+          column: 1,
+          endColumn: 5,
+          line: 4,
+          messageId: noEffectDecoratorAndCreator,
+          suggestions: [
+            {
+              messageId: noEffectDecoratorAndCreatorSuggest as MessageIds,
```

Comment:
Forgot to comment about this... it's interesting that we need this typecast here. If we remove the compiler isn't satisfied anymore...

<img width="662" alt="Error without typecast" src="https://user-images.githubusercontent.com/11965907/132385112-600b02e7-9479-4e76-9f20-76a734bb72c7.png">

It seems like the `RuleTester.run` isn't being able to catch all `MessageIds` properly if it's a union type 🤔 

<img width="766" alt="bug in RuleTester.run?" src="https://user-images.githubusercontent.com/11965907/132385066-ec609844-e7fd-49fb-87aa-936dccb2fc6d.png">